MergeKSorted.java

```diff
@@ -0,0 +1,22 @@
+class Solution {
+    public ListNode mergeKLists(ListNode[] lists) {
+        PriorityQueue<Integer> pq = new PriorityQueue<>();
+        
+        for(ListNode listnode : lists){
+            while(listnode!=null){
+                pq.add(listnode.val);
+                listnode = listnode.next;
+            }
+        }
+        
+        ListNode dummy = new ListNode(-1);
+        ListNode head = dummy;
+        
+        while(!pq.isEmpty()){
+            head.next = new ListNode(pq.remove());
+            head = head.next;
+        }
+        
+        return dummy.next;
+    }
+}
```

Comment:
I don't think this would work. Please look at it again and fix it. Look for the solution on the S30 website if you face trouble solving it by yourself.